src/goto-programs/adjust_float_expressions.cpp

```diff
@@ -118,7 +118,7 @@ void adjust_float_expressions(exprt &expr, const exprt &rounding_mode)
                                 irep_idt());
 
       expr.operands().resize(3);
-      expr.op2()=rounding_mode;
+      to_ieee_float_op_expr(expr).rounding_mode() = rounding_mode;
```

Comment:
Shouldn't these two lines just be combined into `expr.add_to_operands(rounding_mode)`?

Reply:
I actually want to protect/remove .add_to_operands on anything but `multi_ary_exprt`.
The best thing here might be to build that ieee_float_op_expr from scratch, as opposed to morphing one. Will be separate PR.